File: Python/Week 1/Day 2/Core/BankAccount/bank account.py
class BankAccount:
    def __init__(self, int_rate, balance):
        self.int_rate = int_rate
        self.balance = balance
    def deposit(self, amount):
        self.balance += amount
        return self

    def withdraw(self, amount):
        if self.balance > amount:
            self.balance -= amount
        else:
            print("Insufficient funds: Charging a $5 fee")
            "and"
            self.balance -= 5
        return self
    
    def display_account_info(self):
        print(f"Balance {self.balance}")
        return self
    
    def yield_interest(self):
        if self.balance > 0:
            self.balance += self.balance * self.int_rate
        return self
    
account1 = BankAccount(0.01, 500)
account1.deposit(100).deposit(100).deposit(100).withdraw(50).yield_interest().display_account_info()
account2 = BankAccount(0.01, 0)
account2.deposit(1).deposit(100).withdraw(50).withdraw(50).withdraw(50).withdraw(50).yield_interest().display_account_info()

File: Java/week_1/Day_2/Core/BankAccount/BankAccount.java
public class BankAccount {
    private double checkingBalance;
    private double savingsBalance;
    private static int accounts;
    private static double totalMoney;

    // CONSTRUCTOR
    public BankAccount(double checkingBalance, double savingsBalance){
        this.checkingBalance = checkingBalance;
        this.savingsBalance = savingsBalance;
        accounts++;  // Increment the accounts count
        totalMoney += (checkingBalance + savingsBalance);  // Add initial balance to totalMoney
    }

    // GETTERS
    public double getCheckingBalance(){
        return checkingBalance;
    }

    public double getSavingsBalance(){
        return savingsBalance;
    }

    public static int getAccounts(){  // Make the getAccounts method static
        return accounts;
    }

    public static double getTotalMoney(){  // Make the getTotalMoney method static
        return totalMoney;
    }

    // METHODS
    public void deposit(double amount, String accountType){
        // Users should be able to deposit money into their checking or savings account
        if ("checking".equalsIgnoreCase(accountType)) {
            checkingBalance += amount;
        } else if ("savings".equalsIgnoreCase(accountType)) {
            savingsBalance += amount;
        }
        totalMoney += amount;
    }

    public void withdraw(double amount, String accountType){
        // Users should be able to withdraw money from their checking or savings account
        // Do not allow them to withdraw money if there are insufficient funds
        if ("checking".equalsIgnoreCase(accountType) && checkingBalance >= amount) {
            checkingBalance -= amount;
            totalMoney -= amount;
        } else if ("savings".equalsIgnoreCase(accountType) && savingsBalance >= amount) {
            savingsBalance -= amount;
            totalMoney -= amount;
        } else {
            System.out.println("Insufficient funds for withdrawal");
        }
    }

    public double getBalance(){
        // Display total balance for checking and savings of a particular bank account
        return checkingBalance + savingsBalance;
    }
}